src/lib/Form.tsx

```diff
@@ -1,34 +1,71 @@
-import { ReactNode } from "react";
+import { ReactNode, useCallback, useEffect, useRef } from "react";
 import { DeepPartial, FieldValues, FormProvider, Resolver, SubmitHandler, useForm, UseFormReturn } from "react-hook-form";
 import { jsonIsoDateReviver } from "./helpers/dateUtils";
 import { InternalFormContext } from "./context/InternalFormContext";
+import { useDebouncedCallback } from "use-debounce";
 
 interface FormProps<T extends FieldValues> {
   onSubmit: SubmitHandler<T>;
   resolver?: Resolver<T>;
   defaultValues?: DeepPartial<T>;
   requiredFields?: (keyof T)[];
+  autoSubmitOnChangeDelayInMs?: number;
```

Comment:
idea 2: 

There a few more options that the `use-debounce` package exposes. 

Maybe we could also expose this here in an object like `autoSubmitConfig`. 

![image](https://github.com/neolution-ch/react-hook-form-components/assets/92933708/21599d53-1260-4eb6-950f-d399f45443cd)

Reply:
yes sure, added a new interface with all the options